source/elements/oneMKL/source/domains/data_fitting/terms.rst

```diff
@@ -0,0 +1,324 @@
+
+Partition Hints
+^^^^^^^^^^^^^^^
+
+The following are supported:
+
+ - Non-uniform.
+ - Quasi-uniform.
+ - Uniform.
+
+**Syntax**
+
+.. code:: cpp
+
+  enum class partition_hint {
+    non_uniform,
+    quasi_uniform,
+    uniform
+  };
+
+Function Values Hints
+^^^^^^^^^^^^^^^^^^^^^
+
+Let :math:`\left\{ x_i \right\}_{i=1,\cdots,nx}` is a partition,
+:math:`\left\{ f_j(x) \right\}_{j=1,\cdots,ny}` is a vector-valued function.
+Function values are stored in the one-dimensional array with ``nx * ny`` elements.
+2 different layouts are possible: row major and column major.
+
+- For row major layout function values are stored as the following:
+
+  Let :math:`\left\{ f_{j,i} \right\}_{j=1,\cdots,ny, i=1,\cdots,nx}`
+  is the function value that corresponds to the ``i-th`` partition point and the ``j-th`` function.
+
+- For column major:
+
+  Let :math:`\left\{ f_{i,j} \right\}_{i=1,\cdots,nx, j=1,\cdots,ny}`
+  is the function value that corresponds to the ``i-th`` partition point and the ``j-th`` function.
+
+The following hints are supported:
+
+ - Row major.
+ - Column major.
+
+**Syntax**
+
+.. code:: cpp
+
+  enum class function_hint {
+    row_major,
+    col_major
+  };
+
+Coefficients Hints
+^^^^^^^^^^^^^^^^^^
+
+Let :math:`\left\{ x_i \right\}_{i=1,\cdots,nx}` is a partition,
+:math:`\left\{ f_j(x) \right\}_{j=1,\cdots,ny}` is a vector-valued function.
+Let cubic spline should be constructed. It means that it requires 4 coefficients
+per each interpolation interval and function value.
+Cofficients are stored in the one-dimensional array with ``4 * (nx - 1) * ny`` elements.
+
+- For row major:
+
+  Let :math:`\left\{ c_{j,i,k} \right\}_{j=1,\cdots,ny, i=1,\cdots,nx-1, k=1,\cdots,4}`
+  is the coefficient value that corresponds to the ``i-th`` partition point, the ``j-th`` function.
+
+- For column major:
+
+  Let :math:`\left\{ c_{i,j,k} \right\}_{i=1,\cdots,nx-1, j=1,\cdots,ny, k=1,\cdots,4}`
+  is the coefficient value that corresponds to the ``i-th`` partition point, the ``j-th`` function.
+
+The following is supported:
+
+ - row major
+
+**Syntax**
+
+.. code:: cpp
+
+  enum class coefficient_hint {
+    row_major
+  };
+
+Sites Hints
+^^^^^^^^^^^
+
+The following are supported:
+
+ - Non-uniform.
+ - Uniform.
+ - Sorted.
+
+**Syntax**
+
+.. code:: cpp
+
+  enum class site_hint {
+    non_uniform,
+    uniform,
+    sorted
+  };
+
+Interpolation Results Hints
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Let :math:`\left\{ f_j(x) \right\}_{j=1,\cdots,ny}` is a vector-valued function,
+:math:`\left\{ s_i \right\}_{i=1,\cdots,ns}` are sites, ``d`` is a number of derivatives (including interpolation values) that needs to be calculated.
+So, size of memory to store interpolation results is ``nsite * ny * d`` elements.
+
+6 different layouts are possible:
+
+ - functions-sites-derivatives
+
+   Let :math:`\left\{ r_{j,i,k} \right\}_{j=1,\cdots,ny, i=1,\cdots,nsite, k=1,\cdots,d}`
+   is an interpolation result that corresponds to the ``i-th`` site,
```

Comment:
**be** an interpolation result...

Reply:
similar all through document:    "Let xxx be a/an ..." instead of "Let xxx is a/an ..."